test/Elastic.Apm.AspNetCore.Tests/DistributedTracingAspNetCoreTests.cs

```diff
@@ -2,44 +2,41 @@
 using System.Linq;
 using System.Net.Http;
 using System.Reflection;
+using System.Threading;
 using System.Threading.Tasks;
+using Elastic.Apm.Api;
 using Elastic.Apm.DiagnosticSource;
 using Elastic.Apm.EntityFrameworkCore;
 using Elastic.Apm.Tests.Mocks;
 using FluentAssertions;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.DependencyInjection;
 using SampleAspNetCoreApp;
-using SampleAspNetCoreApp.Data;
 using Xunit;
 
 namespace Elastic.Apm.AspNetCore.Tests
 {
-	/// <summary>
-	/// Distributed tracing integration test.
-	/// It starts <see cref="SampleAspNetCoreApp" /> with 1 agent and <see cref="WebApiSample" /> with another agent.
-	/// It calls the 'DistributedTracingMiniSample' in <see cref="SampleAspNetCoreApp" />, which generates a simple HTTP
-	/// response
-	/// by calling <see cref="WebApiSample" /> via HTTP.
-	/// Makes sure that all spans and transactions across the 2 services have the same trace id.
-	/// </summary>
 	[Collection("DiagnosticListenerTest")]
-	public class DistributedTracingAspNetCoreTests : IDisposable
+	public class DistributedTracingAspNetCoreTests : IDisposable, IAsyncLifetime
```

Comment:
We don't have to do this, but just a suggestion: we could remove the `IDisposable` interface and move the 2 lines from `Dispose()` to `DisposeAsync()`, which is called just before `Dispose()`. Additionally the code from the constructor could be also moved to `InitalizeAsync()` , like this:
```
public Task InitializeAsync()
{
  return Task.Factory.StartNew(() =>
  {
    //code from .ctor
  }
}}
```

(one side effect is that some fields can't be readonly).

With this we would have 2 methods less. Again, just a minor thing, won't change much, I just think the code would be simpler.

Reply:
Done